src/main/java/com/gotocompany/depot/http/request/body/TemplatizedJsonBody.java

```diff
@@ -32,60 +28,20 @@ public String build(MessageContainer msgContainer) throws IOException {
         }
-        return parse(jsonObject, parsedMessage).toString();
-    }
 
-    private JSONObject parse(JSONObject object, ParsedMessage parsedMessage) {
-        try {
-            Set<String> keys = object.keySet();
-            JSONObject finalJsonObject = new JSONObject();
-            for (String key : keys) {
-                Object value = object.get(key);
-                Template templateKey = new Template(key);
-                Object parsedKey = templateKey.parseWithType(parsedMessage);
-                if (value instanceof JSONObject) {
-                    finalJsonObject.put(parsedKey.toString(), parse((JSONObject) value, parsedMessage));
-                } else if (value instanceof JSONArray) {
-                    JSONArray tempJsonArray = parseJsonArray((JSONArray) value, parsedMessage);
-                    finalJsonObject.put(parsedKey.toString(), tempJsonArray);
-                } else if (value instanceof String) {
-                    Template templateValue = new Template((String) value);
-                    finalJsonObject.put(parsedKey.toString(), templateValue.parseWithType(parsedMessage));
-                } else {
-                    finalJsonObject.put(parsedKey.toString(), value);
-                }
-            }
-            return finalJsonObject;
-        } catch (InvalidTemplateException e) {
-            throw new IllegalArgumentException(e.getMessage());
-        }
+        JsonNode parsedJsonNode = JsonParserUtils.parse(templateJsonNode, parsedMessage);
+        return parsedJsonNode.toString();
     }
 
-    private JSONArray parseJsonArray(JSONArray jsonArray, ParsedMessage parsedMessage) throws InvalidTemplateException {
-        JSONArray tempJsonArray = new JSONArray();
-        for (int i = 0; i < jsonArray.length(); i++) {
-            Object object = jsonArray.get(i);
-            if (object instanceof JSONObject) {
-                tempJsonArray.put(parse((JSONObject) object, parsedMessage));
-            } else if (object instanceof JSONArray) {
-                tempJsonArray.put(parseJsonArray((JSONArray) object, parsedMessage));
-            } else if (object instanceof String) {
-                Template templateValue = new Template((String) object);
-                tempJsonArray.put(templateValue.parseWithType(parsedMessage));
-            } else {
-                tempJsonArray.put(object);
-            }
-        }
-        return tempJsonArray;
-    }
 
-    private JSONObject createJsonObject(String jsonTemplate) {
+    private JsonNode createJsonNode(String jsonTemplate) {
         if (jsonTemplate.isEmpty()) {
             throw new ConfigurationException("Json body template cannot be empty");
         }
         try {
-            return new JSONObject(jsonTemplate);
-        } catch (JSONException e) {
+            return JsonParserUtils.getObjectMapper().readTree(jsonTemplate);
```

Comment:
we can put this method inside JsonParserUtils. Because getting static variable seems weird.

Reply:
fixed